pkg/config/validation/validation.go

```diff
@@ -2347,6 +2381,30 @@ func ValidateProtocol(protocolStr string) error {
 	return nil
 }
 
+// wrapper around multierror.Append that enforces the invariant that if all input errors are nil, the output
+// error is nil (allowing validation without branching).
+func appendValidation(v Validation, vs ...error) Validation {
```

Comment:
should this just be a member of `Validation`?

Reply:
I considered this but didn't do it because appendErrors doesn't and `append` builtin also does not